src/Compiler/Microsoft.AspNetCore.Razor.Language/test/TestFiles/ParserTests/TagHelperBlockRewriterTest/CreatesMarkupCodeSpans_EscapedExpression_01.stree.txt

```diff
@@ -13,25 +13,22 @@
                     MarkupTextLiteral - [12..13)::1 - ['] - Gen<Markup>
                         SingleQuote;['];
                     MarkupTagHelperAttributeValue - [13..32)::19
-                        MarkupBlock - [13..17)::4
-                            CSharpCodeBlock - [13..17)::4
-                                CSharpImplicitExpression - [13..17)::4
-                                    CSharpTransition - [13..14)::1 - Gen<None>
-                                        Transition;[@];
-                                    CSharpImplicitExpressionBody - [14..17)::3
-                                        CSharpCodeBlock - [14..17)::3
-                                            CSharpExpressionLiteral - [14..17)::3 - [new] - Gen<Expr>
-                                                Keyword;[new];
-                        CSharpExpressionLiteral - [17..28)::11 - [ string("1,] - Gen<None>
-                            Whitespace;[ ];
-                            Text;[string(];
-                            DoubleQuote;["];
-                            Text;[1,];
-                        CSharpExpressionLiteral - [28..32)::4 - [ 2")] - Gen<None>
-                            Whitespace;[ ];
-                            Text;[2];
-                            DoubleQuote;["];
-                            Text;[)];
+                        CSharpCodeBlock - [13..32)::19
+                            CSharpImplicitExpression - [13..32)::19
+                                CSharpTransition - [13..14)::1
+                                    Transition;[@];
+                                CSharpImplicitExpressionBody - [14..32)::18
+                                    CSharpCodeBlock - [14..32)::18
+                                        CSharpExpressionLiteral - [14..32)::18 - [new string("1, 2")] - Gen<None>
+                                            Keyword;[new];
+                                            Whitespace;[ ];
+                                            Text;[string(];
+                                            DoubleQuote;["];
+                                            Text;[1,];
+                                            Whitespace;[ ];
+                                            Text;[2];
+                                            DoubleQuote;["];
+                                            Text;[)];
```

Comment:
This diff in the second commit should demonstrate the change well. Previously for attribute value `@new string("1, 2")`, we had syntax tree with three nodes `MarkupBlock[@new]`, `CSharpExpression[ string("1,]`, `CSharpExpression[ 2")]`. That caused the later codegen to emit `#line` directive for each part separately, breaking the code. I guess the fix could be done at that later stage as well - avoid generating separate `#line` directives for attribute values or something like that. But this felt like the most correct place to do the fix - now the syntax tree is for example equivalent to the one for `@(new string("1, 2"))`.